src/solvers/kriging.jl

```diff
@@ -98,34 +99,30 @@ function preprocess(problem::EstimationProblem, solver::Kriging)
 
       # determine path and neighborhood search method
       if varparams.maxneighbors ≠ nothing
-        # locations with data for variable
-        varlocs = collect(keys(datamap(problem, var)))
-
         if varparams.neighborhood ≠ nothing
           # local search with a neighborhood
           neigh = varparams.neighborhood
 
-          # create a path from the data and outwards
-          # use at most 10^2 points to generate path
-          N = length(varlocs); M = ceil(Int, N/10^2)
-          path = SourcePath(view(varlocs,1:M:N))
-
-          searcher  = NeighborhoodSearcher(pdomain, neigh)
-          bsearcher = BoundedSearcher(searcher, maxneighbors)
+          if neigh isa BallNeighborhood
+            #for specialised knn and ball search combined
+            bsearcher = KBallSearcher(pdata, maxneighbors,neigh)
+          else
+            searcher  = NeighborhoodSearcher(pdata, neigh)
+            bsearcher = BoundedSearcher(searcher, maxneighbors)
+          end
         else
           # nearest neighbor search with a distance
           distance = varparams.distance
           path = LinearPath()
-          bsearcher = NearestNeighborSearcher(pdomain, maxneighbors,
-                                              locations=varlocs, metric=distance)
+          bsearcher = NearestNeighborSearcher(pdata, maxneighbors,metric=distance)
         end
       else
         # use all data points as neighbors
-        path = LinearPath()
```

Comment:
Should we keep the path here for consistency? I noticed you moved it to outside the if-else branch?

Reply:
I believe we just need LinearPath before creating preproc. Although it seems path is not longer needed, for not sure. Maybe this could be done later.

Reply:
Nice, path is also gone :100: it was needed before because of the `SourcePath` trick. Glad we don't need it anymore here.